src/core/iomgr/fd_posix.c

```diff
@@ -172,20 +181,25 @@ void grpc_fd_ref(grpc_fd *fd) { ref_by(fd, 2); }
 void grpc_fd_unref(grpc_fd *fd) { unref_by(fd, 2); }
 
 static void make_callback(grpc_iomgr_cb_func cb, void *arg, int success,
```

Comment:
change signature to take a grpc_iomgr_closure* instead of cb, arg, and then call it directly if allow_synchronous_callback?